src/index.ts

```diff
@@ -86,6 +89,94 @@ const {grpc} = new gax.GrpcClient();
  * @property {Constructor} [promise] Custom promise module to use instead of
  *     native Promises.
  */
+export interface Client {
+  //tslint: disable-next-line: no-any
+  [k: string]: any;
+  servicePath?: string;
+  sslCreds?: ChannelCredentials;
+  port?: number;
+}
+export interface ClientConfig {
+  [k: string]: string|{}|Function|Client|undefined;
+  apiEndpoint?: string;
+  projectId?: string;
+  appProfileId?: string;
+  email?: string;
+  keyFilename?: string;
+  credentials?: {
+    client_email: string;
+    private_key: string;
+  };
+  autoRetry?: boolean;
+  maxRetries?: number;
+  promise?: Constructor<{}>;
+}
+export interface Options extends ClientConfig {
+  BigtableClient?: Client;
+  BigtableInstanceAdminClient?: Client;
+  BigtableTableAdminClient?: Client;
+}
+
+export type CreateInstanceResponse = [IInstance, GaxOperation, LongrunningIOperation];
+export type GetInstanceResponse = [IInstance[], ListInstancesResponse];
+
+export type ICluster = btAdminClient.bigtable.admin.v2.ICluster;
+export type LongrunningIOperation = btAdminClient.longrunning.IOperation;
+export type ListInstancesResponse = btAdminClient.bigtable.admin.v2.IListInstancesResponse;
+export type InstanceState = btAdminClient.bigtable.admin.v2.Instance.State;
+export type InstanceType = btAdminClient.bigtable.admin.v2.Instance.Type;
+
+export interface CreateInstanceCallback {
+  (instance?: Instance, operation?: GaxOperation, apiResponse?: LongrunningIOperation): void;
+}
+export interface GetInstanceCallback {
+  (err?: Error|null, instances?: Instance[], apiResponse?: ListInstancesResponse): void;
+}
+export interface GetProjectIdCallback {
+  (err: Error|null, projectId?: string|null): void;
+}
+
+export interface CreateInstanceCluster {
+  id: string;
+  location: string;
+  nodes: number;
+  storage: string;
+}
+export interface CreateInstanceClusters {
+  [k: string]: {
+    location: string;
+    serveNodes: number;
+    defaultStorageType: ICluster;
+  };
+}
+export interface CreateInstanceOptions {
+  clusters?: CreateInstanceCluster[];
+  displayName?: string;
+  labels?: string|{[k: string]: string};
+  type?: string;
+  gaxOptions?: CallOptions;
+}
+export interface CreateInstanceRequest {
+  parent?: string|null;
+  instanceId?: string|null;
+  clusters?: ({[k: string]: ICluster}|null);
+  instance?: IInstance|null;
+}
+//tslint:disable-next-line interface-name
+export interface IInstance {
```

Comment:
I don't really like the `I` prefix for interfaces, and it isn't something we've used elsewhere.  Is there a proper Instance type somewhere else in the project?

Reply:
This is my fault I think, lol. For exporting generated response types I was at a loss for what to call them.

Reply:
I renamed it `ManualIInstance` if that's okay for now 🤷‍♂️